src/fondant/pipeline/pipeline.py

```diff
@@ -220,12 +220,27 @@ def from_ref(cls, ref: t.Any, **kwargs) -> "ComponentOp":
                 image = ref.image()
                 description = ref.__doc__ or "lightweight component"
 
+                # Try to determine produce for TransformComponents
+                if hasattr(cls, "resolve_produce"):
+                    try:
+                        produces = cls.resolve_produce(consumes=kwargs["consumes"])
```

Comment:
I think we should only attempt to resolve it if no `produces` argument is passed to the component. Would also return valid logs to the user that it is inferred and return the inferred schema